### PR TITLE
fix(grid): DetectChanges when apply column's formatter via code #3819

### LINKED ISSUE #3819: Grid doesn't reflect the applied formatter immediately
### Description  
Grid doesn't reflect the applied column's formatter

 * igniteui-angular version: 7.1.x
 * browser: any

### Steps to reproduce  
1. Run the sample
2. Click the AddSummary button

### Result  
'Salary' cells values are formatted, but that change is not reflected at the UI.
Need to scroll to see the changes

### Expected result  
See the changes immediately without scrolling

### Attachments  
[mainProj.zip](https://github.com/IgniteUI/igniteui-angular/files/2831649/mainProj.zip)

### PR DESCRIPTION
**Pull request by astaev**
Closes #3819 

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid/column.spec.ts [hunk 1]
```diff
@@ -359,3 +391,23 @@ export class ColumnHaederClassesComponent {
     @ViewChild(IgxGridComponent, { read: IgxGridComponent })
     public grid: IgxGridComponent;
 }
+
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/grid/column.spec.ts]: Could we re-use some of the existing components?

File: projects/igniteui-angular/src/lib/grids/column.component.ts [hunk 2]
```diff
@@ -827,6 +839,10 @@ export class IgxColumnComponent implements AfterContentInit {
      *@hidden
      */
     protected _summaries = null;
+    /**
+     *@hidden
+     */
+    protected _formatter = null;
```
[the remark this answers is not in the document]
Comment (astaev) [reply projects/igniteui-angular/src/lib/grids/column.component.ts]: Done

File: projects/igniteui-angular/src/lib/grids/grid/column.spec.ts (hunk 1 above)
Comment (astaev) [reply projects/igniteui-angular/src/lib/grids/grid/column.spec.ts]: Existing component now used

File: projects/igniteui-angular/src/lib/grids/column.component.ts [hunk 3]
```diff
@@ -352,18 +352,30 @@ export class IgxColumnComponent implements AfterContentInit {
...
+     *  return new Intl.DateTimeFormat("en-US").format(val); }
      * ```
      * @memberof IgxColumnComponent
      */
-    @Input()
-    public formatter: (value: any) => any;
+    public set formatter(value: any) {
+        if (this._formatter !== value && this.grid) {
+            this._formatter = value;
+            this.grid.cdr.detectChanges();
```
[the remark this answers is not in the document]
Comment (astaev) [reply projects/igniteui-angular/src/lib/grids/column.component.ts]: Done

**Pull request merged**